GVRf/Extensions/gvrf-particlesystem/src/main/java/org/gearvrf/particlesystem/Particles.java

```diff
@@ -0,0 +1,124 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.particlesystem;
+
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRMaterial;
+import org.gearvrf.GVRMaterialShaderManager;
+import org.gearvrf.GVRMesh;
+import org.gearvrf.GVRRenderData;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.GVRShaderTemplate;
+import org.gearvrf.GVRTexture;
+import org.joml.Vector3f;
+import org.joml.Vector4f;
+
+import static android.opengl.GLES20.GL_POINTS;
+
+/**
+ *  This class is responsible for instancing meshes, with vertices that act as
+ *  the actual particles when rendered using GL_POINTS.
+ */
+
+public class Particles extends GVRMesh {
```

Comment:
Still derives from GVRMesh. Also it looks like it should be package private class in the emitters package. Are end users supposed to create meshes themselves directly?